File: web-application/src/main/resources/public/javascript/ro/ovidiuconeac/foodtavern/webapp/components/Navigation.js

/**
 * Created by ovidiu on 4/3/17.
 */
var Navigation = (function () {

    // Instance stores a reference to the Singleton
    var instance;

    function init() {
        return {

            showFoodUI: function () {
                $('#includedContent').load('resources/layout/FoodView.html');
            },

            showServerConnectionUI: function () {
                $('#includedContent').load('resources/layout/ServerConnectionView.html');
            }
        };

    };

    return {

        // Get the Singleton instance if one exists
        // or create one if it doesn't
        getInstance: function () {

            if ( !instance ) {
                instance = init();
            }

            return instance;
        }

    };

})();

File: web-application/src/main/java/ro/ovidiuconeac/foodtavern/webapp/components/Navigation.java
package ro.ovidiuconeac.foodtavern.webapp.components;

/**
 * Created by ovidiu on 4/10/17.
 */
public class Navigation {
    private static Navigation ourInstance = new Navigation();

    public static Navigation getInstance() {
        return ourInstance;
    }

    private Navigation() {
    }
}
